controllers/helpers.go

```diff
@@ -84,22 +87,54 @@ func findVMByUUID(ctx context.Context, client *nutanixClientV3.Client, uuid stri
 	return response, nil
 }
 
+func generateProviderID(uuid string) string {
+	return fmt.Sprintf("%s%s", providerIdPrefix, uuid)
+}
+
+func getVMUUID(ctx context.Context, nutanixMachine *infrav1.NutanixMachine) (string, error) {
+	vmUUID := nutanixMachine.Status.VmUUID
+	if vmUUID != "" {
+		if _, err := uuid.Parse(vmUUID); err != nil {
+			return "", fmt.Errorf("VMUUID was set but was not a valid UUID: %s err: %v", vmUUID, err)
+		}
+		return vmUUID, nil
+	}
+	providerID := nutanixMachine.Spec.ProviderID
+	if providerID == "" {
+		return "", nil
+	}
+	id := strings.TrimPrefix(providerID, providerIdPrefix)
+	// Not returning error since the ProviderID initially is not a UUID. CAPX only sets the UUID after VM provisioning. If it is not a UUID, continue.
+	if _, err := uuid.Parse(id); err != nil {
+		return "", nil
```

Comment:
why nil when parsing error? unlike above at line 98. can we be consistent here or above